### ISSUE TITLE
Thread.abort_on_exception = true is bad

### ISSUE THREAD

**Issue by jeffreylammers** [phobos/phobos#151]
https://github.com/phobos/phobos/blob/9bbee049a3f04ef5b1f0b92dc58b3162bd2db715/lib/phobos.rb#L32

Why is Thread.abort_on_exception being set here?  This makes this library unusable for me in a multi-threaded program as any error in my thread that uses phobos to publish data causes my server-based program to abort.

Why does this have to be set?

**Comment by jeffreylammers** [comment]
To expand on that - I can trap exceptions in my thread - but if I'm using other threaded libraries - like logging or others - then I have to figure out in all places how to trap errors from those libraries.  This is simply not practical and at best a huge risk.

**Comment by dorner** [comment]
The main reason was when using Phobos as a consumer. We need to be able to abort on exception or we don't have a good way to safely exit when asked. If you have a separate process for consumers vs. your application, you can always set this setting back to false for your app if it's affecting your functionality.

**Comment by jeffreylammers** [comment]
Unfortunately that does not work because other libraries (like logging) get initialized outside of my control and their threads abort and I have no way to control that.  Can't the instance method be used instead so you set that just on the threads you need? e.g. Thread.current.abort_on_exception = true (but only inside your own threads)?

**Comment by dorner** [comment]
Yep, that could definitely work! Can you submit a PR with the fix?

**Comment by jeffreylammers** [comment]
I've not delved into where your threads are created / managed as it would have to be set there.  If you can point me to that part of the code, I can certainly try...

**Comment by dorner** [comment]
.... lol I took a look and we actually *do* set the abort_on_exception to the individual threads! https://github.com/phobos/phobos/blob/master/lib/phobos/executor.rb#L29

I think it might be safe for us to just remove that one line then.

**Comment by jeffreylammers** [comment]
haha - ok - yeah - I just found the spec and in the spec it does it at the instance level - which was a bit confusing...
https://github.com/phobos/phobos/blob/127fd9a9420e0674e35383b2d367a5a75ed54c33/spec/lib/phobos/listener_spec.rb#L59
